javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java

```diff
@@ -165,315 +166,390 @@ private List<ClientMethod> createClientMethods(Operation operation) {
+                }
 
-            List<ClientMethodParameter> parameters = new ArrayList<>();
-            List<String> requiredParameterExpressions = new ArrayList<>();
-            Map<String, String> validateExpressions = new HashMap<>();
-            List<MethodTransformationDetail> methodTransformationDetails = new ArrayList<>();
+                Set<Parameter> originalParameters = new HashSet<>();
+                for (Parameter parameter : codeModelParameters) {
+                    ClientMethodParameter clientMethodParameter = Mappers.getClientParameterMapper().map(parameter);
 
-            List<Parameter> codeModelParameters;
-            if (settings.isDataPlaneClient()) {
-                // Required path and body parameters are allowed
-                codeModelParameters = request.getParameters().stream().filter(p -> p.isRequired() &&
-                        (p.getProtocol().getHttp().getIn() == RequestParameterLocation.PATH ||
-                            p.getProtocol().getHttp().getIn() == RequestParameterLocation.BODY ||
-                            p.getProtocol().getHttp().getIn() == RequestParameterLocation.HEADER ||
-                            p.getProtocol().getHttp().getIn() == RequestParameterLocation.QUERY))
-                    .collect(Collectors.toList());
-            } else {
-                codeModelParameters = request.getParameters().stream().filter(p -> !p.isFlattened()).collect(Collectors.toList());
-            }
+                    if (request.getProtocol() != null
+                            && request.getProtocol().getHttp() != null
+                            && request.getProtocol().getHttp().getMediaTypes() != null
+                            && request.getProtocol().getHttp().getMediaTypes().stream().anyMatch(mediaType -> mediaType.equals(
+                            "application/json-patch+json"))) {
+                        clientMethodParameter = CustomClientParameterMapper.getInstance().map(parameter);
+                    }
 
-            Set<Parameter> originalParameters = new HashSet<>();
-            for (Parameter parameter : codeModelParameters) {
-                ClientMethodParameter clientMethodParameter = Mappers.getClientParameterMapper().map(parameter);
+                    // If the codemodel parameter and proxy method parameter types don't match, update the client
+                    // method param to use proxy method parameter type.
+                    if (clientMethodParameter.getClientType() == GenericType.FluxByteBuffer
+                            && proxyMethod.getParameters().stream().anyMatch(proxyMethodParameter -> proxyMethodParameter.getClientType() == ClassType.BinaryData)) {
+                        clientMethodParameter = updateClientMethodParameter(clientMethodParameter);
+                    }
+                    if (request.getSignatureParameters().contains(parameter)) {
+                        parameters.add(clientMethodParameter);
+                    }
```

Comment:
This code looks a bit hacky. I am OK with it, but do we again need a `toNewBuilder` as `updateClientMethodParameter`?

The change to indent makes it hard to see what is changed. Let me know if there are other important changes in this file (e.g. the one for `ClientMethod.equals`?).